pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -191,23 +191,26 @@ let
     # disabling fortify implies fortify3 should also be disabled
     then lib.unique (hardeningDisable ++ [ "fortify3" ])
     else hardeningDisable;
-  supportedHardeningFlags = [ "fortify" "fortify3" "stackprotector" "pie" "pic" "strictoverflow" "format" "relro" "bindnow" ];
-  # Musl-based platforms will keep "pie", other platforms will not.
-  # If you change this, make sure to update section `{#sec-hardening-in-nixpkgs}`
-  # in the nixpkgs manual to inform users about the defaults.
-  defaultHardeningFlags = if stdenv.hostPlatform.isMusl &&
-      # Except when:
-      #    - static aarch64, where compilation works, but produces segfaulting dynamically linked binaries.
-      #    - static armv7l, where compilation fails.
-      !(stdenv.hostPlatform.isAarch && stdenv.hostPlatform.isStatic)
-    then supportedHardeningFlags
-    else lib.remove "pie" supportedHardeningFlags;
+  knownHardeningFlags = [
```

Comment:
While we're at it, maybe we could also extract hardcoded constants in add-hardening.sh?

https://github.com/NixOS/nixpkgs/blob/8caf3dcf347422e30dafe09534a82b0b62c314d6/pkgs/build-support/cc-wrapper/add-hardening.sh#L28

https://github.com/NixOS/nixpkgs/blob/8caf3dcf347422e30dafe09534a82b0b62c314d6/pkgs/build-support/bintools-wrapper/add-hardening.sh#L18

Reply:
Flippin heck I forgot about those.

Reply:
Honestly think this is too tricky to do without introducing a bizarre two-way dependency between `mkDerivation` and the wrappers. And attempting to bring the `knownHardeningFlags` into the wrappers alongside `defaultHardeningFlags` causes all sorts of problems around bootstrapping and use of unwrapped compilers/binutils because all of a sudden `mkDerivation` has an empty `knownHardeningFlags` which causes it to throw errors if any derivation happens to use `hardeningDisable` or `hardeningEnable`.